src/pam-config.c

```diff
@@ -868,6 +869,20 @@ main (int argc, char *argv[])
 	      opt_set->enable (opt_set, "is_enabled", opt.opt_val);
 	    }
 	  break;
+	case 2500:
+	  /* pam_fscrypt.so */
+	  if (opt.m_query)
+	    print_module_config (common_module_list, "pam_fscrypt.so");
+	  else
+	    {
+	      if (!opt.m_delete && check_for_pam_module ("pam_fscrypt.so", opt.force, 0) != 0)
+		return 1;
+	      opt_set = mod_pam_fscrypt.get_opt_set (&mod_pam_fscrypt, AUTH);
+	      opt_set->enable (opt_set, "is_enabled", opt.opt_val);
+	      opt_set = mod_pam_fscrypt.get_opt_set (&mod_pam_fscrypt, SESSION);
+	      opt_set->enable (opt_set, "is_enabled", opt.opt_val);
+	    }
+	  break;
```

Comment:
This part looks wrong, pam_fscrypt.go seems to need the password type, too, but this is missing here.
Since mod_pam_fscrypt.c would do it correct, why do you overwrite it here?

Reply:
> This part looks wrong, pam_fscrypt.go seems to need the password type, too, but this is missing here.

Good catch, fixed.

> Since mod_pam_fscrypt.c would do it correct, why do you overwrite it here?

I do not understand the comment. what am I overwriting? I just copy&pasted the code from the other module section, what specific did I miss? it appears to work here (apparently I'm not testing it well enough)

Reply:
The whole pam-config.c section is only needed if you need to overwrite the default settings of mod_pam_fscrypt.c or need to do special things not possible with the default.
In your case, the full pam-config.c change could be dropped and it should just work.